refactor(oxlint): pass `DiagnosticService` as a parameter for `TsGoLintState.lint()`

Sysix opened the pull request:
Started to refactor for the language server, but noticed it needs more to be done: https://github.com/oxc-project/tsgolint/issues/71
Still opened the PR to align with the service:

https://github.com/oxc-project/oxc/blob/ca91a261556d88cde7ece4314d905a78aca1cc76/crates/oxc_linter/src/service/mod.rs#L91-L102

camc314 commented:
### Merge activity

* **Aug 11, 7:06 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(oxlint): pass `DiagnosticService` as a parameter for `TsGoLintState.lint()` (#13004)

Started to refactor for the language server, but noticed it needs more to be done: https://github.com/oxc-project/tsgolint/issues/71
Still opened the PR to align with the service:

https://github.com/oxc-project/oxc/blob/ca91a261556d88cde7ece4314d905a78aca1cc76/crates/oxc_linter/src/service/mod.rs#L91-L102
